articles/3_sass-selectors.md

```diff
@@ -0,0 +1,237 @@
+
+### Added Specificity
+
+A common use case for the `&` is to add specificity under certain cases. For example, say you have a div that always has the class `.demo`, but under certain cases (determined by your javascript code), it also has the class `.special`. You want the `div` to always be styled using the `.demo` CSS but sometimes you want `.special` styles *also*.
+
+See the [codepen demo](http://codepen.io/raemadeline/pen/ZObaba) for a real example. In codepen, you can toggle between viewing the SCSS and the compiled CSS, so you can see how your code compiles :)
+
+You might have noticed that this demo would have worked just as well if you never nested the styles
+
+```
+.demo {
+  /* demo styles */
+}
+
+.special {
+  /* special styles */
+}
+```
+
+And within that example, this is true. There's no need (and often its not desired) to style on a more specific selector than a more general one (stay tuned for my blog post on specificity!). Sometimes it is needed, so I'm showing this demo more as a proof of concept than as a guide for how things should *always* be done.
+
+### Pseudo-Classes
+
+If you had noticed the styling on the button in the [codepen demo](http://codepen.io/raemadeline/pen/ZObaba) above, you'd see that I used the `&` selector to nest the pseudo-selectors for the button within the styling for the button.
+
+```
+button {
+  /* normal styles */
+
+  &:hover {
+    /* hover styles */
+  }
+}
+```
+
+Compiles into
+
+```
+button {
+  /* normal styles */
+}
+
+button:hover {
+  /* hover styles */
+}
+```
+
+Sure, you could just write out the compiled CSS and it would still work, but from a readability standpoint it is easier to keep track of code when it is nested well into organized components.
+
+### Modifier Classes
+
+The ampersand does not just concatenate selectors, it can also concatenate words to form selectors. Suppose you're building a complex design system that includes several different buttons, all of which have their own classes (currently in Addepar, we have `btn`, `btn-link`, `btn-icon`, and several others).
+
+You can write SCSS to keep all these styles together and well organized:
+
+```
+.btn {
+  /* base button styling here */
+
+  &-link {
+    /* styling for buttons that should look like links */
+  }
+
+  &-icon {
+    /* styling for buttons with icons in them */
+  }
+}
+```
+
+And as you could probably guess, this compiles nicely into:
+
+```
+.btn {
+  /* base button styling here */
+}
+
+.btn-link {
+  /* styling for buttons that should look like links */
+}
+
+.btn-icon {
+  /* styling for buttons with icons in them */
+}
+```
+
+This feature is especially useful if you're working on a project that uses [BEM notation](http://getbem.com/introduction/), which is a very useful system of organization, but we do not use it within Addepar so I won't cover it in depth here.
```

Comment:
Again, I would exclude discussing Addepar's specific practices. The link and brief description alone seem like they'd be good enough for anyone who wants to learn more.